overlord/hookstate/ctlcmd/netplanapply.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+var shortNetplanApplyHelp = i18n.G("The netplan-apply command applies network configuration from netplan.")
+var longNetplanApplyHelp = i18n.G("TODO")
+
+func init() {
+	addCommand("netplan-apply", shortNetplanApplyHelp, longNetplanApplyHelp, func() command {
+		return &netplanApplyCommand{}
+	})
+}
+
+type netplanApplyCommand struct {
+	baseCommand
+}
+
+// Execute runs the netplan-apply command after confirming the calling snap has
+// permissions to run "netplan apply" with network-setup-control connected and
+// the netplan-apply attribute specified on the plug
+func (c *netplanApplyCommand) Execute(args []string) error {
+	ctx := c.context()
+	if ctx == nil {
+		return errors.New("missing snapctl context")
+	}
+
+	// check if netplan apply can be used with this context
+	canUse, err := canUseNetplanApply(ctx)
+	if err != nil {
+		return err
+	}
+	if !canUse {
+		// TODO: is there a better error type to return here?
+		return fmt.Errorf("cannot use netplan apply - must have network-setup-control interface connected with netplan-apply attribute specified as true")
+	}
+
+	// note: don't lock state here, we lock/unlock it inside netplanApplyTaskSet
+	st := ctx.State()
+	if st == nil {
+		return errors.New("context state is nil")
+	}
+
+	// create new netplan apply task set
+	tts, err := netplanApplyTaskSet(st, ctx)
+	if err != nil {
+		return err
+	}
+
+	// lock the state and create a new change for the task
+	st.Lock()
+	chg := st.NewChange("netplan-apply", fmt.Sprintf("Applying netplan network configuration for snap %q", ctx.InstanceName()))
+	for _, ts := range tts {
+		chg.AddAll(ts)
+	}
+	st.EnsureBefore(0)
+	st.Unlock()
+
+	// wait for the change to be ready or for it to timeout
+	select {
+	case <-chg.Ready():
+		st.Lock()
+		defer st.Unlock()
+		return chg.Err()
+	case <-time.After(configstate.ConfigureHookTimeout() / 2):
+		return errors.New("netplan apply command is taking too long")
+	}
+}
+
+func canUseNetplanApply(ctx *hookstate.Context) (bool, error) {
+	// note: the following is adapted from devicestate.CanManageRefreshes
+	st := ctx.State()
+	if st == nil {
+		return false, errors.New("context state is nil")
+	}
+
+	// lock the state and get all snap states
+	st.Lock()
+	defer st.Unlock()
+
+	var snapst snapstate.SnapState
+	if err := snapstate.Get(st, ctx.InstanceName(), &snapst); err != nil {
+		return false, err
+	}
+
+	// Always get the current info even if the snap is currently
+	// being operated on or if its disabled.
+	info, err := snapst.CurrentInfo()
+	if err != nil {
+		return false, err
+	}
+	if info.Broken != "" {
+		return false, err
+	}
+
+	// TODO: should a check for a snap snap declaration from the store be
+	// here too?
+
+	for _, plugInfo := range info.Plugs {
+		if plugInfo.Interface == "network-setup-control" {
+			attrVal, ok := plugInfo.Attrs["netplan-apply"]
+			if !ok {
+				return false, nil
+			}
+			switch attrVal {
```

Comment:
see browser_support.go, this can/should be an actual bool, not a string

Reply:
This is fixed and is done the same way as browser_support.go now